src/pybind/mgr/cephadm/inventory.py

```diff
@@ -946,6 +964,15 @@ def get_draining_hosts(self) -> List[HostSpec]:
             h for h in self.mgr.inventory.all_specs() if '_no_schedule' in h.labels
         ]
 
+    def get_conf_keyring_draining_hosts(self) -> List[HostSpec]:
+        """
+        Returns all hosts that have _no_conf_keyring label and therefore should have
+        no config files or client keyring placed on them, but are potentially still reachable
+        """
+        return [
+            h for h in self.mgr.inventory.all_specs() if '_no_conf_keyring' in h.labels
+        ]
+
```

Comment:
shouldn't this function check also for the `_no_schedule` label? the other `get_draining_hosts` function does check for this label. If it shouldn't then maybe we should remove the `draining` from the function name? WDYT

Reply:
no, the whole idea here is to separate out draining the daemons from a host with draining the conf and keyring to allow doing one but not the other. If we check `_no_schedule` here then the label that causes the daemons to be drained will also cause the conf keyrings to be drained which is how it worked before and what this PR is trying to change.

Reply:
I see, I thought that `draining` in this context was refering to daemons as well. In this case we are just draining the conf.